Tests/NFUnitTestStringBuilder/StringPerformanceTests.cs

```diff
@@ -0,0 +1,1071 @@
+        {
+            RunTest(StringTests.LastIndexOf3);
+        }
+
+        [TestMethod]
+        public void LastIndexOf_Test_4()
+        {
+            RunTest(StringTests.LastIndexOf4);
+        }
+
+        [TestMethod]
+        public void LastIndexOf_Test_5()
+        {
+            RunTest(StringTests.LastIndexOf5);
+        }
+
+        [TestMethod]
+        public void Length_Test()
+        {
+            RunTest(StringTests.Length);
+        }
+
+        [TestMethod]
+        public void Split_Test_0()
+        {
+            RunTest(StringTests.Split0);
+        }
+
+        [TestMethod]
+        public void Split_Test_1()
+        {
+            RunTest(StringTests.Split1);
+        }
+
+        [TestMethod]
+        public void SubString_Test_0()
+        {
+            RunTest(StringTests.SubString0);
+        }
+
+        [TestMethod]
+        public void SubString_Test_1()
+        {
+            RunTest(StringTests.SubString1);
+        }
+
+        [TestMethod]
+        public void ToCharArray_Test_0()
+        {
+            RunTest(StringTests.ToCharArray0);
+        }
+
+        [TestMethod]
+        public void ToCharArray_Test_1()
+        {
+            RunTest(StringTests.ToCharArray1);
+        }
+
+        [TestMethod]
+        public void ToLower_Test()
+        {
+            RunTest(StringTests.ToLower);
+        }
+
+        [TestMethod]
+        public void ToString_Test()
+        {
+            RunTest(StringTests.ToString);
+        }
+
+        [TestMethod]
+        public void ToUpper_Test()
+        {
+            RunTest(StringTests.ToUpper);
+        }
+
+        [TestMethod]
+        public void Trim_Test_0()
+        {
+            RunTest(StringTests.Trim0);
+        }
+
+        [TestMethod]
+        public void Trim_Test_1()
+        {
+            RunTest(StringTests.Trim1);
+        }
+
+        [TestMethod]
+        public void TrimEnd_Test()
+        {
+            RunTest(StringTests.TrimEnd);
+        }
+
+        [TestMethod]
+        public void TrimStart_Test()
+        {
+            RunTest(StringTests.TrimStart);
+        }
+        [TestMethod]
+        public void Test1()
+        {
+            Assert.True(true);
+        }
+
+
+        private void RunTest(StringTests test)
+        {
+            long start = 0, stop = 0;
+            int maxLength = 3000;
+            int iterations = 10;
+            int[] time = new int[iterations];
+            int currLength = 1;
+            double duration = 0;
+
+            while (currLength < maxLength + 1)
+            {
+                for (int i = 0; i < iterations; i++)
+                {
+                    switch (test)
+                    {
+                        case StringTests.Ctor0:
+                            String str0;
+                            char[] token00 = GetTokenCharArray(currLength);
+                            start = DateTime.UtcNow.Ticks;
+                            str0 = new String(token00);
```

Comment:
and using the benchmark tool, that's basically the line you'd like to test and so you'll have to include the measurement there. And rather than running only once, maybe better to run it 100 times or so to get a better understanding of the all up performance

Reply:
any screen capture/output you can share? So we can have an idea on how it looks like? Code wise seems ok.

Reply:
![image](https://user-images.githubusercontent.com/1071761/196458011-2f945f73-4d07-45fc-9b81-652a8cd2c017.png)

Reply:
It then definitely seems the right direction :-) How do you like the tool?

Reply:
Its good.  Biggest complaint is I have to run against real device.  Would be much nicer to run like unit tests which don't require.